sw/device/silicon_creator/rom_ext/rom_ext_start.S

```diff
@@ -157,6 +165,27 @@ _rom_ext_start_boot:
   la   t0, (_rom_ext_interrupt_vector + 1)
   csrw mtvec, t0
 
+  /**
+   * Initialize the Ibex simple address translation to run position dependent
+   * code on the virtual flash space.
+   */
+  .extern _flash_start
+  .extern _vflash_start
+  .extern _vflash_size
+  la a0, _vflash_start
+  la a1, _vflash_size
+  la a2, _flash_start
+  call _ibex_address_translation_set  
+
+  /**
+   * Configure the epmp to enable read and execution on the virtual flash 
+   * space.
+   */
+  la a0, _vflash_start
+  la a1, _vflash_size
+  li a2, (EPMP_CFG_L | EPMP_CFG_R | EPMP_CFG_X)
+  call _epmp_configure_napot
+
```

Comment:
In an ideal world I think we'd jump to virtual flash here.

Reply:
Do you mean forcing executing on the virtual address after epmp setup?
if so, the jump to `rom_ext_main` would remain the same.

Reply:
Yes. So we'd jump to another assembly function at this point which sets up the C runtime and then jumps to `rom_ext_main`.

Reply:
Done